index.ts

```diff
@@ -0,0 +1,18 @@
+interface IEventEmitter<T = any> {
+  on(event: 'data', listener: (...args: [T, ...any[]]) => void): this;
+  on(event: string | symbol, listener: (...args: any[]) => void): this;
+}
+
+export function promisifyEventEmitter(event: IEventEmitter): Promise<undefined>;
+export function promisifyEventEmitter<T>(event: IEventEmitter, result: T): Promise<T>;
+export function promisifyEventEmitter<T>(event: IEventEmitter, result?: T | undefined): Promise<T | undefined> {
```

Comment:
While we need this functionality here, it feels a bit wrong to explicitly export it here.
I have the feeling that we might end up using the `arrayify-stream` package every time we need to promisify an event emitter, which is probably not what we want.

What we could do instead is just copy `promisifyEventEmitter` to a new package, and remove the `export` keyword here? (I would not add a dependency here, to keep this package as lightweight as possible).

Reply:
New package made here https://github.com/jeswr/promisify-event-emitter - unfortunately I release it under my namespace due to a heavier package of the same name already exisitng

Reply:
Perhaps another name can be used instead? I see `event-emitter-promisify` and `event-emitter-promise` are still free.